lib/config-win32.h

```diff
@@ -612,6 +612,14 @@ Vista
 #  endif
 #endif
 
+/* Define whether the CERT_CHAIN_ENGINE_CONFIG structure has support for
+ * setting exclusive trust stores. This is used to implement the CAINFO easy
+ * option for Schannel.
+ */
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0601)
+#define HAVE_EXCLUSIVE_TRUST_MODE 1
```

Comment:
Interesting. Perhaps the CERT_CHAIN_ENGINE_CONFIG definition has changed in more recent versions of the Windows SDK?

Here's what the definition looks like in the 8.0 SDK:
```
typedef struct _CERT_CHAIN_ENGINE_CONFIG {

    DWORD       cbSize;
    HCERTSTORE  hRestrictedRoot;
    HCERTSTORE  hRestrictedTrust;
    HCERTSTORE  hRestrictedOther;
    DWORD       cAdditionalStore;
    HCERTSTORE* rghAdditionalStore;
    DWORD       dwFlags;
    DWORD       dwUrlRetrievalTimeout;      // milliseconds
    DWORD       MaximumCachedCertificates;
    DWORD       CycleDetectionModulus;

#if (NTDDI_VERSION >= NTDDI_WIN7)
    HCERTSTORE  hExclusiveRoot;
    HCERTSTORE  hExclusiveTrustedPeople;
#endif

#if (NTDDI_VERSION >= NTDDI_WIN8)
    DWORD       dwExclusiveFlags;
#endif

} CERT_CHAIN_ENGINE_CONFIG, *PCERT_CHAIN_ENGINE_CONFIG;
```

When I explicitly set _WIN32_WINNT to 0x0600 (Windows Server 2008), the hExclusiveRoot field is not present in the structure. Given this, I don't think this check can be done at runtime.

But, I do notice that other ifdef checks specific to Schannel are in the schannel.c file, for example, HAS_ALPN. Given that, I can move this check into schannel.c (or schannel_verify.c) instead of having it live in config-win32.h.